### ISSUE TITLE
Support arbitrary ES client API calls in tracks

### ISSUE THREAD

**Issue by danielmitterdorfer** [elastic/rally#105]
Currently we only support selected API calls in tracks. This leads to problems when we need to call unsupported APIs (see e.g. elastic/rally-tracks#3). We should find a way to support more (and ideally arbitrary) API calls in Rally.

**Comment by danielmitterdorfer** [comment]
Concrete idea by @ywelsch: allow to start primary relocation from node A to B and back. This obviously makes only sense after we've implemented #71 (or if users set up their own cluster and use the `--benchmark-only` pipeline). So it might make sense to define a separate ticket for that later.

**Comment by danielmitterdorfer** [comment]
Another use case: evaluate reindex performance (suggested by @nik9000). For this we'd also need a more flexible approach for tracks, as we want to vary the number of shards in the target index.

**Comment by danielmitterdorfer** [comment]
#107 laid the groundwork for this task by allowing arbitrary parameter sources via a plugin API. We will not support arbitrary calls in Rally core but we will provide a new plugin extension point which allows users to write custom "runners" for a track.

**Comment by danielmitterdorfer** [comment]
Rally allows now create any API calls that the Python Elasticsearch client supports. Here is a short example (based on the fix for elastic/rally-tracks#3):

In track.json specify an operation with type "percolate" (you can choose this name freely):

```json
    {
      "name": "percolator_with_content_google",
      "operation-type": "percolate",
      "body": {
        "doc": {
          "body": "google"
        },
        "track_scores": true
      }
    }
```

Then create a file `track.py` next to `track.json` and implement the following two functions 

```python
def percolate(es, params):
    es.percolate(
        index="queries",
        doc_type="content",
        body=params["body"]
    )
    return 1, "ops"


def register(registry):
    registry.register_runner("percolate", percolate)
```

Details are in the [user docs](http://esrally.readthedocs.io/en/latest/adding_tracks.html#custom-runners).
